src/java/picard/illumina/IlluminaBasecallsConverter.java

```diff
@@ -453,8 +453,8 @@ public synchronized void addRecord(final String barcode, final CLUSTER_OUTPUT_RE
 
         private synchronized SortingCollection<CLUSTER_OUTPUT_RECORD> newSortingCollection() {
             final int maxRecordsInRam =
-                    maxReadsInRamPerTile /
-                            barcodeRecordWriterMap.size();
+                    Math.max(1, maxReadsInRamPerTile /
```

Comment:
Not ideal, but if someone is setting the number of reads in memory to be that low, then spillage to disk will ensure. I'm giving enough rope to make the program run (slowly perhaps) without barfing (what would happen if 0 was allowed). This isn't the problem that lead me here in the first place. the actual fix is the un-commented line.